jsonrpc/server.go

```diff
@@ -145,8 +154,16 @@ func (s *Server) RegisterMethod(method Method) error {
 	if handlerT.Kind() != reflect.Func {
 		return errors.New("handler must be a function")
 	}
-	if handlerT.NumIn() != len(method.Params) {
-		return errors.New("number of function params and param names must match")
+	var needsContext bool
+	numArgs := handlerT.NumIn()
+	if numArgs > 0 {
+		if handlerT.In(0).Implements(contextInterface) {
+			numArgs--
+			needsContext = true
```

Comment:
I believe you can directly set `method.needsContext` here, there is no need to a separate  `var needsContext bool`. `method` is a copy anyways.